### DISCUSSION TITLE
Is it possible to rewrite these asserts?

### DISCUSSION THREAD

**Discussion by JasonGrace2282** [pytest-dev/pytest#12776] [Q&A]
Does anybody know of a hook that is able to rewrite something like below?
```py
assert val
```
(`pytest_assertrepr_compare` only works with stuff like `assert x == y`)

Alternatively, is it possible to rewrite asserts like
```py
assert request.status_code == 400
```
to something like
```py
assert request.status_code == 400, f"Incorrect status code for {request}
```
With `pytest_assertrepr_compare`, you only get access to the value after it's evaluated, which is annoying.

**Comment by dongfangtianyu** [comment]
As far as I know, pytest's assertion rewriting is relatively complex in detail. 
The purpose of this rewriting might be to display detailed differences based on the simple use of the `assert` keyword.

---

I didn't quite understand what your specific need is, does it need to be compatible with the original assert statement?
but if you can avoid using the `assert`  directly and instead use a function that contains `assert`, 
it will give you much more flexibility, which might meet your various complex needs for assertions.
